### PR TITLE
cpu/saml21: fix RTC wrong condition for #if

### PR DESCRIPTION
**Pull request by dylad**
### Contribution description

This PR fixes a wrong #if condition for the RTC driver.

### CODE REVIEW

**Comment by kaspar030** [comment]
How does this manifest? (How can I test?)

**Comment by dylad** [comment]
This condition was written because SAML21 revision B must set this bit if you want to use `rtc_get_time` function. 
SAML21 revision A don't need this bit (probably because a sillicon bug between this two variants). This issue is descripted in this [document](http://ww1.microchip.com/downloads/en/AppNotes/Atmel-42512-Migrating-from-SAM-L21-Variant-A-to-Variant-B_ApplicationNote_AT13382.pdf)
Since SAML21-XPRO board can be equipped with either rev A (old boards not sold anymore but they are still in the wild) or rev B, this bit must be set for rev B (to use `rtc_get_time` function).
I don't know if we can set this bit with a SAML21 rev A so I decide to keep the #if but just change the condition because the previous one cannot occur as SAML21XXXB does not exist in RIOT.

So if you want to test it, you will need a one SAML21 rev B and  one rev A :)

File: cpu/saml21/periph/rtc.c [hunk 1]
```diff
@@ -97,7 +97,7 @@ void rtc_init(void)
 
     /* RTC config with RTC_MODE2_CTRL_CLKREP = 0 (24h) */
     RTC->MODE2.CTRLA.reg = RTC_MODE2_CTRLA_PRESCALER_DIV1024 |  /* CLK_RTC_CNT = 1KHz / 1024 -> 1Hz */
-#if (SAML21XXXB) || (SAMR30)
+#if (CPU_MODEL_SAML21J18B)
```
Comment (kaspar030) [inline cpu/saml21/periph/rtc.c]: How about changing this like in this patch:

https://gist.github.com/9842253687bd2698a9b344779dd83500

E.g., define the bit if not visible, and always set it.
I can confirm that setting the bit on SAML21 rev A doesn't hurt.

Comment (dylad) [reply cpu/saml21/periph/rtc.c]: > I can confirm that setting the bit on SAML21 rev A doesn't hurt

Then I'll change it :)

**Comment by dylad** [comment]
@kaspar030 I've pushed the requested changes.

**Comment by kaspar030** [comment]
pls squash!

**Comment by dylad** [comment]
Squashed

**Pull request merged**